### ISSUE TITLE
coco datasets  is wrongfully implemented.

### ISSUE THREAD

**Issue by sailfish009** [open-mmlab/mmdetection#3465]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
1. i have tested mmdetection(DetectoRS version) and latest mmdetection. 
mmdet/datasets/{coco.py, custom.py}
 2. DetectoRS shows normal train accuracy, but latest mmdetection shows 100 accuracy at first epoch

**Reproduction**
1. What command or script did you run?
```
$ python tools/train.py configs/DetectoRS/DetectoRS_mstrain_400_1200_r50_40e.py    # in DetectoRS root directory
$ python tools/train.py configs/detectors/cascade_rcnn_r50_rfp_1x_coco.py                    # in mmdetection root directory
```
2. Did you make any modifications on the code or config? Did you understand what you have modified?
this is simple, just replaced latest coco.py and custom.py with old version from DetectoRS. 

3. What dataset did you use?
kaggle global wheat dataset

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
sys.platform: linux
Python: 3.7.8 (default, Jun 29 2020, 05:44:46) [GCC 7.5.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GPU 0: GeForce RTX 2080 Ti
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.5.1
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute  _61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PY
  TORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-t  ype-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-  strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old  -style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format   -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON,   USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.1
OpenCV: 4.2.0
MMCV: 1.0.4
MMDetection: 2.3.0rc0+c342781
MMDetection Compiler: GCC 7.5
MMDetection CUDA Compiler: 10.1

2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [ pip ]
  

**Bug fix**

workaround is simple, just replace coco.py and  custom.py [mmdet/datasets/{coco.py, custom.py}] with older version from DetectoRS(mmdetection v1)

**Comment by ZwwWayne** [comment]
Since mmdetection v1 and mmdetection v2 has many difference as indicated [here](https://github.com/open-mmlab/mmdetection/blob/master/docs/compatibility.md). You may need to modify your dataset accordingly. Both `coco.py` and `custom.py` are tested with many models, they are used to train all the models in the model zoo and the models' performance is normal.

**Comment by ZwwWayne** [comment]
This PR is closed because the implementation of `coco.py` and `custom.py` has no problem. If you have any reimplementation issues, please use the `reimeplementation questions` template to create a new issue and describe your problem specifically and precisely.

**Comment by sailfish009** [comment]
@ZwwWayne  the dataset was based on mmdetection v2.
https://www.kaggle.com/m1nglei/mmdetection-v2-detectors-training-colab
